# Analisis de la Felicidad 

In [3]:
import csv
import pandas as pd

In [5]:
felicidad = pd.read_csv("felicidad.csv")
felicidad.head(10)

pais  anio  escalera_vida   log_pib  soporte_social  \
0  Afghanistán  2008       3.723590  7.168690        0.450662   
1  Afghanistán  2009       4.401778  7.333790        0.552308   
2  Afghanistán  2010       4.758381  7.386629        0.539075   
3  Afghanistán  2011       3.831719  7.415019        0.521104   
4  Afghanistán  2012       3.782938  7.517126        0.520637   
5  Afghanistán  2013       3.572100  7.522238        0.483552   
6  Afghanistán  2014       3.130896  7.516955        0.525568   
7  Afghanistán  2015       3.982855  7.500539        0.528597   
8  Afghanistán  2016       4.220169  7.497038        0.559072   
9  Afghanistán  2017       2.661718  7.497755        0.490880   

   expectativa_vida  libertad  generosidad  percepcion_corrupcion  \
0         50.799999  0.718114     0.177889               0.881686   
1         51.200001  0.678896     0.200178               0.850035   
2         51.599998  0.600127     0.134353               0.706766   
3         51.919998  0.495901     0.172137               0.731109   
4         52.240002  0.530935     0.244273               0.775620   
5         52.560001  0.577955     0.070403               0.823204   
6         52.880001  0.508514     0.113184               0.871242   
7         53.200001  0.388928     0.089091               0.880638   
8         53.000000  0.522566     0.051365               0.793246   
9         52.799999  0.427011    -0.112198               0.954393   

   afecto_positivo  afecto_negativo  confianza  calidad_democracia  \
0         0.517637         0.258195   0.612072           -1.929690   
1         0.583926         0.237092   0.611545           -2.044093   
2         0.618265         0.275324   0.299357           -1.991810   
3         0.611387         0.267175   0.307386           -1.919018   
4         0.710385         0.267919   0.435440           -1.842996   
5         0.620585         0.273328   0.482847           -1.879709   
6         0.531691         0.374861   0.409048           -1.773257   
7         0.553553         0.339276   0.260557           -1.844364   
8         0.564953         0.348332   0.324990           -1.855426   
9         0.496349         0.371326   0.261179           -1.886566   

   calidad_entrega  de_escalera_pais_anio  gini_banco_mundial  \
0        -1.655084               1.774662                 NaN   
1        -1.635025               1.722688                 NaN   
2        -1.617176               1.878622                 NaN   
3        -1.616221               1.785360                 NaN   
4        -1.404078               1.798283                 NaN   
5        -1.403036               1.223690                 NaN   
6        -1.312503               1.395396                 NaN   
7        -1.291594               2.160618                 NaN   
8        -1.392713               1.796219                 NaN   
9        -1.437808               1.454051                 NaN   

   gini_banco_mundial_promedio  
0                          NaN  
1                          NaN  
2                          NaN  
3                          NaN  
4                          NaN  
5                          NaN  
6                          NaN  
7                          NaN  
8                          NaN  
9                          NaN

## Calculo del promedio y desviacion estandar  de la felicidad para el anio 2006

In [14]:
pAnio = felicidad[felicidad['anio'] == 2006]
pAnio_avg = pAnio['escalera_vida'].mean()
print('El promedio de felicidad para el anio 2006 es:',pAnio_avg)

pdvAnio = felicidad[felicidad['anio'] == 2006]
panio_dv = pdvAnio['escalera_vida'].std()
print('La desviacion estandar de felicidad para el anio 2006 es:',panio_dv)

El promedio de felicidad para el anio 2006 es: 5.1969352893615035
La desviacion estandar de felicidad para el anio 2006 es: 1.0945578179795974


## Calculo del promedio y desviacion estandar de la felicidad para el anio 2018

In [16]:
uAnio = felicidad[felicidad['anio'] == 2018]
uAnio_avg = uAnio['escalera_vida'].mean()
print('El promedio de felicidad para el anio 2018 es:',uAnio_avg)

udvAnio = felicidad[felicidad['anio'] == 2018]
uanio_dv = udvAnio['escalera_vida'].std()
print('La desviacion estandar de felicidad para el anio 2018 es:',uanio_dv)

El promedio de felicidad para el anio 2018 es: 5.502134340650895
La desviacion estandar de felicidad para el anio 2018 es: 1.1034612436939357


## Indice de felicidad de Ecuador anios 2006 y 2018

In [31]:
ecuador1 = felicidad[(felicidad['pais'] == 'Ecuador') & (felicidad['anio'] == 2006)]
felicidadEcua1 = ecuador1['escalera_vida'].values
print('La felicidad en Ecuador para el 2006 fue:', felicidadEcua1)

ecuador2 = felicidad[(felicidad['pais'] == 'Ecuador') & (felicidad['anio'] == 2018)]
felicidadEcua2 = ecuador2['escalera_vida'].values
print('La felicidad en Ecuador para el 2018 fue:', felicidadEcua2)



La felicidad en Ecuador para el 2006 fue: [5.02419138]
La felicidad en Ecuador para el 2018 fue: [6.12801027]


# Indice de felicidad de India anios 2006 y 2018

In [32]:
india1 = felicidad[(felicidad['pais'] == 'India') & (felicidad['anio'] == 2006)]
felicidadIndia1 = india1['escalera_vida'].values
print('La felicidad en India para el 2006 fue:', felicidadIndia1)

india2 = felicidad[(felicidad['pais'] == 'India') & (felicidad['anio'] == 2018)]
felicidadIndia2 = india2['escalera_vida'].values
print('La felicidad en India para el 2018 fue:', felicidadIndia2)


La felicidad en India para el 2006 fue: [5.34825897]
La felicidad en India para el 2018 fue: [3.81806874]


## Comparacion del indice de generosidad de Ecuador e India vs resto del mundo

In [36]:
promedio_generosidad = felicidad['generosidad'].mean()
print("Promedio de generosidad mundial:", promedio_generosidad)


Promedio de generosidad mundial: 7.86347701148664e-05


In [40]:
ecuador3 = felicidad[(felicidad['pais'] == 'Ecuador')]
generosidad_ecuador= ecuador3['generosidad'].mean()
print('El promedio de generosidad en Ecuador es:', generosidad_ecuador)

El promedio de generosidad en ecuador es: -0.11193826450751375


In [43]:
india3 = felicidad[(felicidad['pais'] == 'India')]
generosidad_india= india3['generosidad'].mean()
print('El promedio de generosidad en India es:', generosidad_india)

El promedio de generosidad en India es: -0.0045392049166063505


Tras el análisis de datos, se evidencia que en el año 2006, Ecuador registró un índice de felicidad de 5.02, en contraste con el índice de la India, que fue de 5.34. El promedio global para ese año fue de 5.19, situando a Ecuador por debajo de la media, mientras que la India superó el promedio mundial.

Por otra parte, el promedio global de felicidad para el anio 2018 se situo en 5.50, el de ecuador para ese anio fue para 6.12 ubicando a este pais por encima del primedio mundial, mientras que para india en este anio desciende drasticamente a 3.81. Por lo tanto Ecuador es mas feliz que la India

Se ha tomado como referencia el analisis de la columna generosidad, en la que se ha realizado un promedio de todos los anios de Ecuador e India respectivamente y se comparon con el promedio mundial. El promedio mundial de generosidad es de  0.0000786 aprox, subsiguiente a este se ubica India con  -0.0045 y en tercer lugar Ecuador con  -0.1119, siendo las personas de la india mas generosas que las de Ecuador. 